packages/ui-react/src/components/Card/Card.tsx

```diff
@@ -98,7 +100,7 @@ function Card({
     >
       {!featured && !disabled && (
         <div className={styles.titleContainer}>
-          <h3 className={classNames(styles.title, { [styles.loading]: loading })}>{title}</h3>
+          {React.createElement(`h${headingLevel}`, { className: `${styles.title} ${loading && styles.loading}` }, title)}
```

Comment:
Minor (doesn't make a difference, but a bit cleaner IMO because of JSX):

```tsx
const Heading = `h${headingLevel}`

return (
  // ...
  <Heading className={classNames(styles.title, { [styles.loading]: loading })}>{title}</h3>
```

Reply:
As discussed, this solution wasn't entirely applicable and since this is an optimisation, we deemed it unnecessary to implement at this moment.